scanner/metadata/taglib/taglib_wrapper.cpp

```diff
@@ -58,15 +64,82 @@ int taglib_read(const FILENAME_CHAR_T *filename, unsigned long id) {
     }
   }
 
+  TagLib::ID3v2::Tag *id3Tags = NULL;
+
   // Get some extended/non-standard ID3-only tags (ex: iTunes extended frames)
   TagLib::MPEG::File *mp3File(dynamic_cast<TagLib::MPEG::File *>(f.file()));
   if (mp3File != NULL) {
-    if (mp3File->ID3v2Tag()) {
-      const auto &frameListMap(mp3File->ID3v2Tag()->frameListMap());
+    id3Tags = mp3File->ID3v2Tag();
+  }
+
+  if (id3Tags == NULL) {
+    TagLib::RIFF::WAV::File *wavFile(dynamic_cast<TagLib::RIFF::WAV::File *>(f.file()));
+    if (wavFile != NULL && wavFile->hasID3v2Tag()) {
+      id3Tags = wavFile->ID3v2Tag();
+    }
+  }
+
+  if (id3Tags == NULL) {
+    TagLib::RIFF::AIFF::File *aiffFile(dynamic_cast<TagLib::RIFF::AIFF::File *>(f.file()));
+    if (aiffFile && aiffFile->hasID3v2Tag()) {
+      id3Tags = aiffFile->tag();
+    }
+  }
 
-      for (const auto &kv : frameListMap) {
-        if (!kv.second.isEmpty())
-          tags.insert(kv.first, kv.second.front()->toString());
+  // Yes, it is possible to have ID3v2 tags in FLAC. However, that can cause problems
+  // with many players, so they will not be parsed
+
+  if (id3Tags != NULL) {
+    const auto &frames = id3Tags->frameListMap();
+
+    for (const auto &kv: frames) {
+      if (kv.first == "USLT") {
+        for (const auto &tag: kv.second) {
+          TagLib::ID3v2::UnsynchronizedLyricsFrame *frame = dynamic_cast<TagLib::ID3v2::UnsynchronizedLyricsFrame *>(tag);
+          if (frame == NULL) continue;
+
+          tags.erase("LYRICS");
+
+          char lyricsTag[11];
+          strncpy(lyricsTag, LYRICS_KEY, 7);
+          strncpy(lyricsTag+7, frame->language().data(), 3);
```

Comment:
What happens if there is no language? Remember that exceptions in the C++ code causes a SIGSEGV in Navidrome and it crashes without a chance to recover. Cant we just send the lyrics to the Go wrapper, and put this logic there? 

I feel very nervous increasing our reliance on C++ code like that.

Reply:
This is **required** to be a 3-byte string (and from taglib source, unless the object is internally modified it is ensured to be three bytes). But for safety, I've added a length check, defaulting to `xxx` otherwise and moved the concatenation to golang.